core/src/test/scala/com/pingcap/tispark/datasource/UpsertSuite.scala

```diff
@@ -44,26 +43,14 @@ class UpsertSuite extends BaseDataSourceSuite("test_datasource_upsert") {
     testSelect(dbtableInSpark, Seq(row1, row2, row3, row4))
 
     // deduplicate=false
-    // insert row5 row5
-    {
-      val caught = intercept[TiBatchWriteException] {
-        batchWrite(List(row5, row5))
-      }
-      assert(
-        caught.getMessage
-          .equals("data conflicts! set the parameter deduplicate.")
-      )
-    }
-
-    // deduplicate=true
-    // insert row5 row5
-    batchWrite(List(row5, row5), Some(Map("deduplicate" -> "true")))
-    testSelect(dbtableInSpark, Seq(row1, row2, row3, row4, row5))
+    // a table does not need to check duplicate if it does not have a primary key
```

Comment:
pls add a test: write data to a table with primary key (int type)

Reply:
will do so.